### PR TITLE
Scripts-repo and path

### PR DESCRIPTION
**Pull request by Agrawal-Anamika**
### Description

- Adding a parameter as scripts repo where all the build and deploy scripts can be placed
- By default it points to application repo and "scripts folder"

---

### Testing Done

---

### CODE REVIEW

File: .build_pipeline/pipeline.yaml [hunk 1]
```diff
@@ -47,18 +47,18 @@ spec:
...
     - name: cos-api-key
       description: The API Key to interact with the Cloud Object Storage Instance
     - name: insights
       description: The variable storing DevOps Insight integration with the toolchain
-    - name: runtime
-      description: The variable storing language runtime 
-    - name: pipeline-repo-branch
-      description: The branch of the pipeline-repo
+    - name: scripts-subpath
+      description: TThe subpath to the script repo from root
```
Comment (akshaykumar-vijapur) [inline .build_pipeline/pipeline.yaml]: scripts repo. missed "s"

**Review by akshaykumar-vijapur — APPROVED** [review]

File: .build_pipeline/pipeline.yaml [hunk 2]
```diff
@@ -124,7 +124,8 @@ spec:
             # to the needs of your build system. For interpreter based project (Python/JS), where the application needs only the source code, you may
             # skip this step or comment out the existing lines and move over to copy the contents to /artifacts/binaries folder.            
             echo "Running Application Source Code Build Script"
-            source ./pipeline-repo/scripts/$(params.runtime)/build.sh
+            subpath="$(params.scripts-subpath)"
```
Comment (mundrapiyush36) [inline .build_pipeline/pipeline.yaml]: @anamikaagrawal123 Make and Try with subpath as SCRIPTS_SUBPATH

File: .build_pipeline/pipeline.yaml (hunk 1 above)
Comment (Agrawal-Anamika) [reply .build_pipeline/pipeline.yaml]: done

File: .build_pipeline/task.yaml [hunk 3]
```diff
@@ -9,8 +9,8 @@ spec:
       default: repository
     - name: appname
       description: The name of the application passed as environment variable
-    - name: runtime
-      description: The application lanugage runtime  
+    - name: scripts-subpath
+      description: The subpath to the script repo from root
```
Comment (mundrapiyush36) [inline .build_pipeline/task.yaml:13]: @anamikaagrawal123  The subpath to the script folder from the base of the git repository storing scripts folder. By default the folder path is scripts

File: .build_pipeline/task.yaml [hunk 4]
```diff
@@ -29,15 +29,16 @@ spec:
...
+        - name: scripts-subpath
+          value: $(params.scripts-subpath) 
       script: |          
           #!/bin/bash
            # This step creates an archive/zip of your artifacts (source code, binaries, images, configuration files etc.) placed in the
            # /output/binaries folder to prepare for the upload in the subsequent step. While it creates the archive/zip, it also calculates
            # the checksum of the archived/zipped contents which will be part of the build metadata to be validated during the download.
           set -e -o pipefail;
-          source ./pipeline-repo/scripts/${RUNTIME}/build-artifact-info.sh
+          subpath="$(params.scripts-subpath)"
```
Comment (mundrapiyush36) [inline .build_pipeline/task.yaml:40]: @anamikaagrawal123 Rename the variable to SCRIPTS_SUBPATH

File: .deploy_pipeline/listener.yaml [hunk 5]
```diff
@@ -64,10 +64,10 @@ spec:
       default: ""
     - name: insights
       description: The variable storing DevOps Insight integration with the toolchain  
-    - name: runtime
-      description: The variable storing language runtime 
-    - name: pipeline-repo-branch
-      description: The branch of the pipeline-repo
+    - name: scripts-subpath
+      description: TThe subpath to the script repo from root
```
Comment (mundrapiyush36) [inline .deploy_pipeline/listener.yaml:68]: @anamikaagrawal123 The subpath to the script folder from the base of the git repository storing scripts folder. By default the folder path is scripts

File: .deploy_pipeline/pipeline.yaml [hunk 6]
```diff
@@ -57,10 +57,10 @@ spec:
       default: ""
     - name: insights
       description: The variable storing DevOps Insight integration with the toolchain
-    - name: runtime
-      description: The variable storing language runtime 
-    - name: pipeline-repo-branch
-      description: The branch of the pipeline-repo
+    - name: scripts-subpath
+      description: The variable storing language scripts-subpath
```
Comment (mundrapiyush36) [inline .deploy_pipeline/pipeline.yaml:61]: @anamikaagrawal123 The subpath to the script folder from the base of the git repository storing scripts folder. By default the folder path is scripts

File: .deploy_pipeline/task.yaml [hunk 7]
```diff
@@ -213,7 +213,8 @@ spec:
         cd /output
         WORKDIR=/home/${HOST_USER_NAME}/app
         BACKUPDIR=${WORKDIR}_backup
-        source ./pipeline-repo/scripts/${RUNTIME}/setup-remote-deploy.sh
+        subpath="$(params.scripts-subpath)"
```
Comment (mundrapiyush36) [inline .deploy_pipeline/task.yaml:216]: @anamikaagrawal123 Same change here as above

File: .deploy_pipeline/task.yaml [hunk 8]
```diff
@@ -240,11 +241,12 @@ spec:
...
         - name: INVENTORY_REPO
           value: $(params.inventory-repo)
-        - name: RUNTIME
-          value: $(params.runtime)            
+        - name: scripts-subpath
+          value: $(params.scripts-subpath)            
       script: |
         #!/bin/bash
-        source ./pipeline-repo/scripts/${RUNTIME}/health-check.sh
+        subpath="$(params.scripts-subpath)"
```
Comment (mundrapiyush36) [inline .deploy_pipeline/task.yaml:248]: @anamikaagrawal123  Same change as above

File: .deploy_pipeline/task.yaml [hunk 9]
```diff
@@ -272,11 +274,12 @@ spec:
...
           value: $(params.inventory-repo)
-        - name: RUNTIME
-          value: $(params.runtime)            
+        - name: scripts-subpath
+          value: $(params.scripts-subpath)            
       script: |
         #!/bin/bash
-        source ./pipeline-repo/scripts/${RUNTIME}/acceptance-tests.sh
+        subpath="$(params.scripts-subpath)"
+        source ./scripts-repo/${subpath}/acceptance-tests.sh
```
Comment (mundrapiyush36) [inline .deploy_pipeline/task.yaml:282]: @anamikaagrawal123 Same change as above

File: .deploy_pipeline/task.yaml [hunk 10]
```diff
@@ -331,17 +334,18 @@ spec:
...
-        - name: RUNTIME
-          value: $(params.runtime) 
+        - name: scripts-subpath
+          value: $(params.scripts-subpath) 
       script: |
         #!/bin/bash
         echo "DEPLOY_STATUS=$DEPLOY_STATUS"
         echo "HEALTH_CHECK_STATUS=$HEALTH_CHECK_STATUS"
         echo "ACCEPTANCE_TEST_STATUS=$ACCEPTANCE_TEST_STATUS"
+        subpath="$(params.scripts-subpath)"
```
Comment (mundrapiyush36) [inline .deploy_pipeline/task.yaml:344]: @anamikaagrawal123 Same change as above

File: .deploy_pipeline/task.yaml [hunk 11]
```diff
@@ -331,17 +334,18 @@ spec:
...
         if [ $DEPLOY_STATUS == "Failed" ] || [ $HEALTH_CHECK_STATUS == "Failed"  ] || [ $ACCEPTANCE_TEST_STATUS == "Failed"  ]; then
-          source ./pipeline-repo/scripts/${RUNTIME}/rollback-deploy.sh
+          source ./scripts-repo/${subpath}/rollback-deploy.sh
```
Comment (mundrapiyush36) [inline .deploy_pipeline/task.yaml:346]: @anamikaagrawal123 Same change as above

File: .pr_pipeline/pr-pipeline.yaml [hunk 12]
```diff
@@ -61,29 +61,30 @@ spec:
...
+      runAfter: [git-clone-scripts-repo]
       taskRef:
         name: tester-run-tests
       params: 
         - name: tests-script
           value: |
             #!/bin/bash            
             echo "Running Unit Test Script"
-            source ./pipeline-repo/scripts/$(params.runtime)/unit-tests.sh
+            subpath="$(params.scripts-subpath)"
```
Comment (mundrapiyush36) [inline .pr_pipeline/pr-pipeline.yaml:86]: @anamikaagrawal123 Same change as above

File: scripts/golang/build.sh [hunk 13]
```diff
@@ -4,7 +4,7 @@
 ######################################################################################
 
 # set the GO run enviroment
-source ./pipeline-repo/scripts/golang/go-run-time.sh
+source ./scripts-repo/${subpath}/go-run-time.sh
```
Comment (mundrapiyush36) [inline scripts/golang/build.sh]: @anamikaagrawal123 Same change as above

File: scripts/golang/package-build.sh [hunk 14]
```diff
@@ -17,7 +17,7 @@
 # based on your selection during toolchain creation.
 
 # set the GO run rnviroment
-source ./pipeline-repo/scripts/golang/go-run-time.sh
+source ./scripts-repo/${subpath}/go-run-time.sh
```
Comment (mundrapiyush36) [inline scripts/golang/package-build.sh]: @anamikaagrawal123 Same change as above

File: scripts/golang/perform-cleanup.sh [hunk 15]
```diff
@@ -25,5 +25,5 @@ else
     exit 1;
 fi
 
-$SSH_CMD ssh $SSH_ARG -o StrictHostKeyChecking=no $HOST_USER_NAME@$VIRTUAL_SERVER_INSTANCE  env RESTOREFILE=$RESTOREFILE HOST_USER_NAME=$HOST_USER_NAME  'bash -s' < ./pipeline-repo/scripts/golang/cleanup.sh
+$SSH_CMD ssh $SSH_ARG -o StrictHostKeyChecking=no $HOST_USER_NAME@$VIRTUAL_SERVER_INSTANCE  env RESTOREFILE=$RESTOREFILE HOST_USER_NAME=$HOST_USER_NAME  'bash -s' < ./scripts-repo/${subpath}/cleanup.sh
```
Comment (mundrapiyush36) [inline scripts/golang/perform-cleanup.sh]: @anamikaagrawal123 Same change as above

File: scripts/golang/rollback-deploy.sh [hunk 16]
```diff
@@ -24,7 +24,7 @@ else
     exit 1;
 fi
 echo "Tasks has failed, Performing Rollback."
-$SSH_CMD ssh $SSH_ARG -o StrictHostKeyChecking=no $HOST_USER_NAME@$VIRTUAL_SERVER_INSTANCE  env RESTOREFILE=$RESTOREFILE HOST_USER_NAME=$HOST_USER_NAME  'bash -s' < ./pipeline-repo/scripts/golang/restore.sh
+$SSH_CMD ssh $SSH_ARG -o StrictHostKeyChecking=no $HOST_USER_NAME@$VIRTUAL_SERVER_INSTANCE  env RESTOREFILE=$RESTOREFILE HOST_USER_NAME=$HOST_USER_NAME  'bash -s' < ./scripts-repo/${subpath}/restore.sh
```
Comment (mundrapiyush36) [inline scripts/golang/rollback-deploy.sh]: @anamikaagrawal123 Same change as above

File: scripts/golang/setup-remote-deploy.sh [hunk 17]
```diff
@@ -30,7 +30,7 @@ else
 fi
 
 BUILDDIR=/home/${HOST_USER_NAME}/${PIPELINERUNID}
-DEPLOY_SCRIPT_PATH="./pipeline-repo/scripts/golang/deploy.sh"
+DEPLOY_SCRIPT_PATH="./scripts-repo/${subpath}/deploy.sh"
```
Comment (mundrapiyush36) [inline scripts/golang/setup-remote-deploy.sh]: @anamikaagrawal123 Same change as above

File: scripts/golang/setup-remote-deploy.sh [hunk 18]
```diff
@@ -42,7 +42,7 @@ echo "Extract the new artifacts in the host machine."
 $SSH_CMD ssh $SSH_ARG  -o StrictHostKeyChecking=no $HOST_USER_NAME@$VIRTUAL_SERVER_INSTANCE "cd ${BUILDDIR} && tar -xf ${OBJECTNAME} && rm ${OBJECTNAME} "
 
 echo "Creating the symlink to the build directory.."
-$SSH_CMD ssh $SSH_ARG -o StrictHostKeyChecking=no $HOST_USER_NAME@$VIRTUAL_SERVER_INSTANCE env PIPELINERUNID=$PIPELINERUNID  HOST_USER_NAME=$HOST_USER_NAME 'bash -s' < ./pipeline-repo/scripts/golang/backup.sh
+$SSH_CMD ssh $SSH_ARG -o StrictHostKeyChecking=no $HOST_USER_NAME@$VIRTUAL_SERVER_INSTANCE env PIPELINERUNID=$PIPELINERUNID  HOST_USER_NAME=$HOST_USER_NAME 'bash -s' < ./scripts-repo/${subpath}/backup.sh
```
Comment (mundrapiyush36) [inline scripts/golang/setup-remote-deploy.sh]: @anamikaagrawal123 Same change as above

File: scripts/golang/unit-tests.sh [hunk 19]
```diff
@@ -6,7 +6,7 @@
 ######################################################################################
 
 # set the GO run enviroment
-source ./pipeline-repo/scripts/golang/go-run-time.sh
+source ./scripts-repo/${subpath}/go-run-time.sh
```
Comment (mundrapiyush36) [inline scripts/golang/unit-tests.sh]: @anamikaagrawal123 Same change as above

File: scripts/java/perform-cleanup.sh [hunk 20]
```diff
@@ -20,5 +20,5 @@ else
     exit 1;
 fi
 
-$SSH_CMD ssh $SSH_ARG -o StrictHostKeyChecking=no $HOST_USER_NAME@$VIRTUAL_SERVER_INSTANCE  env RESTOREFILE=$RESTOREFILE HOST_USER_NAME=$HOST_USER_NAME  'bash -s' < ./pipeline-repo/scripts/java/cleanup.sh
+$SSH_CMD ssh $SSH_ARG -o StrictHostKeyChecking=no $HOST_USER_NAME@$VIRTUAL_SERVER_INSTANCE  env RESTOREFILE=$RESTOREFILE HOST_USER_NAME=$HOST_USER_NAME  'bash -s' < ./scripts-repo/${subpath}/cleanup.sh
```
Comment (mundrapiyush36) [inline scripts/java/perform-cleanup.sh]: @anamikaagrawal123 Same change as above

File: scripts/java/rollback-deploy.sh [hunk 21]
```diff
@@ -20,7 +20,7 @@ else
     exit 1;
 fi
 echo "Tasks has failed, Performing Rollback."
-$SSH_CMD ssh $SSH_ARG -o StrictHostKeyChecking=no $HOST_USER_NAME@$VIRTUAL_SERVER_INSTANCE  env RESTOREFILE=$RESTOREFILE HOST_USER_NAME=$HOST_USER_NAME  'bash -s' < ./pipeline-repo/scripts/java/restore.sh
+$SSH_CMD ssh $SSH_ARG -o StrictHostKeyChecking=no $HOST_USER_NAME@$VIRTUAL_SERVER_INSTANCE  env RESTOREFILE=$RESTOREFILE HOST_USER_NAME=$HOST_USER_NAME  'bash -s' < ./scripts-repo/${subpath}/restore.sh
```
Comment (mundrapiyush36) [inline scripts/java/rollback-deploy.sh]: @anamikaagrawal123 Same change as above

File: scripts/java/setup-remote-deploy.sh [hunk 22]
```diff
@@ -28,7 +28,7 @@ else
 fi
 
 BUILDDIR=/home/${HOST_USER_NAME}/${PIPELINERUNID}
-DEPLOY_SCRIPT_PATH="./pipeline-repo/scripts/golang/deploy.sh"
+DEPLOY_SCRIPT_PATH="./scripts-repo/${subpath}/deploy.sh"
```
Comment (mundrapiyush36) [inline scripts/java/setup-remote-deploy.sh]: @anamikaagrawal123 Same change as above

File: scripts/java/setup-remote-deploy.sh [hunk 23]
```diff
@@ -40,7 +40,7 @@ echo "Extract the new artifacts in the host machine."
 $SSH_CMD ssh $SSH_ARG  -o StrictHostKeyChecking=no $HOST_USER_NAME@$VIRTUAL_SERVER_INSTANCE "cd ${BUILDDIR} && tar -xf ${OBJECTNAME} && rm ${OBJECTNAME} "
 
 echo "Creating the symlink to the build directory.."
-$SSH_CMD ssh $SSH_ARG -o StrictHostKeyChecking=no $HOST_USER_NAME@$VIRTUAL_SERVER_INSTANCE env PIPELINERUNID=$PIPELINERUNID  HOST_USER_NAME=$HOST_USER_NAME 'bash -s' < ./pipeline-repo/scripts/java/backup.sh
+$SSH_CMD ssh $SSH_ARG -o StrictHostKeyChecking=no $HOST_USER_NAME@$VIRTUAL_SERVER_INSTANCE env PIPELINERUNID=$PIPELINERUNID  HOST_USER_NAME=$HOST_USER_NAME 'bash -s' < ./scripts-repo/${subpath}/backup.sh
```
Comment (mundrapiyush36) [inline scripts/java/setup-remote-deploy.sh]: @anamikaagrawal123 Same change as above

**Review by mundrapiyush36 — CHANGES_REQUESTED** [review]

**Review by mundrapiyush36 — APPROVED** [review]

**Pull request merged**